### DISCUSSION TITLE
Bump messages on master as version source

### DISCUSSION THREAD

**Discussion by dauthleikr** [GitTools/GitVersion#4325] [Q&A] [answered]
We have a master branch and feature branches. We want to use bump-messages only, and tags optionally. 

When a feature branch is created, we want to create nugets with a prerelease version (depeding on the used bump message). This part works.

When a feature branch is merged back to master, a pipeline runs which releases the nuget (master = release branch) with the full version. 

Problem scenario:
1. Package on master is 1.0.0
2. Feature branch is created -> commit says to increment the minor version
3. Prerelease package is built (1.1.0-blablubb)
4. Change is reviewed and merged back into master
5. Release package is built (1.1.0)
6. The next feature branch is created -> commit says to increment the minor version
7. We expect the package to have 1.2.0, but it is still 1.1.0

My understanding is this is because the bump messages on master do not count as a version source. It thinks "there are two minor changes on top of 1.0.0, so 1.1.0 should be fine to describe both changes". How can I signal that 1.1.0 is already released, and can be used as a base version, **WITHOUT USING A TAG** (we want the process to be fully automated)

**Comment by HHobeck** [comment, marked as answer]
Please try the following `TrunkBased` workflow configuration (gitversion.yaml):

```yaml
workflow: TrunkBased/preview1
```

Please notice that the merge no fast forward option needs to be used when merging from the feature to the main branch (the merge message needs to contain the source branch).
